codex/merkletree/codex/coders.nim

```diff
@@ -100,3 +102,18 @@ proc decode*(_: type CodexProof, data: seq[byte]): ?!CodexProof =
       nodes.add node
 
   CodexProof.init(mcodec, index.int, nleaves.int, nodes)
+
+proc fromJson*(
+  _: type CodexProof,
+  json: JsonNode
+): ?!CodexProof =
+  expectJsonKind(Cid, JString, json)
+  var bytes: seq[byte]
+  try:
+    bytes = hexToSeqByte(json.str)
+  except ValueError as err:
+    return failure(err)
+
+  CodexProof.decode(bytes)
+
+func `%`*(proof: CodexProof): JsonNode = % byteutils.toHex(proof.encode())
```

Comment:
This whole change includes changing of the type of quota usage (used, reserved and available bytes) from simple `uint` to `NBytes`.  Then to avoid converting it to `uint` everywhere I changed whenever it was suitable also to NBytes, that includes the data model used for REST endpoint, the `RestRepoStore` object type. And since it's need to be serialized properly on the endpoint we need such encoder.